test/extended/prometheus/prometheus.go

```diff
@@ -42,6 +42,145 @@ import (
 	helper "github.com/openshift/origin/test/extended/util/prometheus"
 )
 
+var _ = g.Describe("[sig-instrumentation][Late] Alerting rules", func() {
+	defer g.GinkgoRecover()
+
+	var url, bearerToken string
+	oc := exutil.NewCLIWithoutNamespace("prometheus")
+
+	g.BeforeEach(func() {
+		var ok bool
+		url, _, bearerToken, ok = helper.LocatePrometheus(oc)
+		if !ok {
+			e2e.Failf("Prometheus could not be located on this cluster, failing prometheus test")
+		}
+	})
+
+	g.It("should comply with OpenShift alerting guidelines", func() {
+		errorFmt := func(group, alert string) func(string, ...interface{}) string {
+			return func(format string, a ...interface{}) string {
+				return fmt.Sprintf("alerting rule %q (group: %s) "+format,
+					append([]interface{}{alert, group}, a...)...)
+			}
+		}
+
+		alertingRules, err := fetchAlertingRules(oc, url, bearerToken)
+		if err != nil {
+			e2e.Fail(err.Error())
+		}
+
+		unexpectedViolations := sets.NewString()
+		severityRe := regexp.MustCompile("^critical|warning|info$")
+
+		for group, alerts := range alertingRules {
+			for _, alert := range alerts {
+				// The Watchdog alert is special because it is only there to
+				// test the end-to-end alert flow and it isn't meant to be
+				// routed to cluster admins.
+				if alert.Name == "Watchdog" {
+					continue
+				}
+
+				alertError := errorFmt(group, alert.Name)
+
+				g.By("having a valid severity label")
+				severity, found := alert.Labels["severity"]
+				if !found {
+					unexpectedViolations.Insert(
+						alertError("should have a 'severity' label"),
+					)
+					continue
+				}
+
+				if !severityRe.MatchString(severity) {
+					unexpectedViolations.Insert(
+						alertError("has a 'severity' label value of %q which doesn't match %q",
+							severity, severityRe.String()),
+					)
+				}
+
+				g.By("having description and summary annotations")
+				if _, found := alert.Annotations["description"]; !found {
+					unexpectedViolations.Insert(
+						alertError("has no 'description' annotation"),
+					)
+				}
+
+				if _, found := alert.Annotations["summary"]; !found {
+					unexpectedViolations.Insert(
+						alertError("has no 'summary' annotation"),
+					)
+				}
+
+				g.By("having a runbook_url annotation if the alert is critical")
+				if _, found := alert.Annotations["runbook_url"]; severity == "critical" && !found {
```

Comment:
if the `runbook_url` annotation is present, it could check that it points to a valid URL.

Reply:
Done. Checks for valid URL and that we can actually fetch it.